src/main/java/meteordevelopment/meteorclient/systems/modules/world/AutoSign.java

```diff
@@ -6,42 +6,293 @@
+        .description("Text in fourth line.")
+        .onChanged(strings -> recompile(back, backScripts, 4))
+        .renderer(StarscriptTextBoxRenderer.class)
+        .visible(backEnabled::get)
+        .build()
+    );
+
+    // Other settings
+    public final Setting<Boolean> overwrite = other.add(new BoolSetting.Builder()
+        .name("overwrite-existing")
+        .description("Overwrite existing text on signs, when possible.")
+        .defaultValue(false)
+        .build());
+
+    private final Setting<String> date = other.add(new StringSetting.Builder()
+        .name("date-format")
+        .description("Format string for {date} object.")
+        .defaultValue("yyyy-MM-dd")
+        .build()
+    );
+
+    private final Setting<String> time = other.add(new StringSetting.Builder()
+        .name("time-format")
+        .description("Format string for {time} object.")
+        .defaultValue("HH:mm z")
+        .build()
+    );
+
+    private final Setting<Boolean> utcTimezone = other.add(new BoolSetting.Builder()
+        .name("UTC-timezone")
+        .description("Whether to use UTC or your local timezone.")
+        .defaultValue(true)
+        .build()
+    );
 
     @Override
-    public void onDeactivate() {
-        text = null;
-    }
+    public WWidget getWidget(GuiTheme theme) {
+        WHorizontalList list = theme.horizontalList();
 
-    @EventHandler
-    private void onSendPacket(PacketEvent.Send event) {
-        if (!(event.packet instanceof UpdateSignC2SPacket)) return;
+        WButton starscriptDoc = theme.button("Open Starscript documentation.");
+        starscriptDoc.action = () -> Util.getOperatingSystem().open("https://github.com/MeteorDevelopment/meteor-client/wiki/Starscript");
+        list.add(starscriptDoc);
+
+        WButton simpleDateFormatDoc = theme.button("Open SimpleDateFormat documentation.");
+        simpleDateFormatDoc.action = () -> Util.getOperatingSystem().open("https://docs.oracle.com/javase/8/docs/api/java/text/SimpleDateFormat.html");
+        list.add(simpleDateFormatDoc);
 
-        text = ((UpdateSignC2SPacket) event.packet).getText();
+        return list;
+    }
+
+    public AutoSign() {
+        super(Categories.World, "auto-sign", "Automatically writes signs. Configure what to write dynamically using Starscript placeholders.");
     }
 
     @EventHandler
-    private void onOpenScreen(OpenScreenEvent event) {
-        if (!(event.screen instanceof AbstractSignEditScreen) || text == null) return;
+    private void onOpenScreen(OpenScreenEvent event) throws InterruptedException {
+        if (!(event.screen instanceof AbstractSignEditScreen)) {return;}
+        if (applyingBack) {
+            applyingBack = false;
+            event.cancel();
+            return;
+        }
 
+        // apply custom format and timezone, not the nicest (temporary overwriting global date and time format)
+        try {
+            StandardLib.dateFormat.applyPattern(date.get());
```

Comment:
i would rather the module not modify global state like that, it would be better to modify `MeteorStarscript` to add specialized time & date functions that accept format strings & timezone identifiers as arguments

Reply:
True.

I saw that there is a [pull requests](https://github.com/MeteorDevelopment/starscript/pull/21) which could resolve this issue.